### ISSUE TITLE
Compile error

### ISSUE THREAD

**Issue by sumo961** [ianharrigan/hxWidgets#22]
Hi,

I'm trying to compile on Mac (10.10.5). I installed wxwidgets 3.0.1 via macports, but I'm not sure that;s the issue. Compile starts, but then stops with an error:

Error: ./src/hx/widgets/StaticBox.cpp:59:14: error: no viable overloaded '='
                this->_ref = boxRef;
                ~~~~~~~~~~ ^ ~~~~~~
/usr/lib/haxe/lib/hxcpp/3,2,205/include/cpp/Pointer.h:235:21: note: candidate function not viable: no known conversion from 'Reference<wxStaticBox>' to 'const Reference<wxWindow>' for 1st argument
   inline Reference operator=( const Reference &inRHS ) { return ptr = inRHS.ptr; }
                    ^
1 error generated.

Any idea why this happens?

**Comment by ianharrigan** [comment]
You need the git version of hxcpp currently.

**Comment by sumo961** [comment]
Installed current github version of hxcpp. Now I'm getting this:

Error: ./src/hx/widgets/GLCanvas.cpp:47:24: error: invalid operands to binary expression ('Array<int>' and 'null')
                bool tmp2 = (options != null());                HX_STACK_VAR(tmp2,"tmp2");
                             ~~~~~~~ ^  ~~~~~~

**Comment by ianharrigan** [comment]
Right, seems like a separate issue. What happens if you remove GLCanvas and any references? Also, what OS are you using? 

Had a bit of a pain implementing GLCanvas a while back so probably related to that. Will check tomorrow on a VM when i get a moment.

Cheers,
Ian

**Comment by sumo961** [comment]
Will try removing GLCanvas later - I'm on OS X 10.10.5

Cheers,

H.

**Comment by sumo961** [comment]
I removed references to GLCanvas. Now I get:

./src/haxe/Resource.cpp:35:20: error: invalid operands to binary expression ('Array<unsigned char>' and 'null')
        bool tmp = (array == null());           HX_STACK_VAR(tmp,"tmp");
                    ~~~~~ ^  ~~~~~~

**Comment by ibilon** [comment]
That's a hxcpp error, I've been investigating and there's a couple things with the latest hxcpp that don't work well with the codebase.

Thankfully haxe 3.3 rc1 will be out soon and that means a haxelib release of hxcpp (and so not having to be compatible with all commits of in progress dev).

**Comment by ianharrigan** [comment]
So hxWidgets **isnt** working with latest hxcpp? :/

**Comment by ibilon** [comment]
More like latest hxcpp has some issues, but by extension yes.

(Note that I don't blame hxcpp, we are using a dev version after all)

**Comment by ianharrigan** [comment]
Right, fair enough, good to know. Are there open issues on hxcpp related to these? Presumably they arent hxWidgets issues only.

**Comment by ibilon** [comment]
I got two pull request merged,

and opened two issues:
one for the array == null https://github.com/HaxeFoundation/hxcpp/issues/404
and another on casting https://github.com/HaxeFoundation/hxcpp/issues/406

**Comment by sumo961** [comment]
So is there a specific hxcpp version that works?

**Comment by ianharrigan** [comment]
im in `3.2.205`

**Comment by ibilon** [comment]
You can do `git checkout 024c25f0690d99445f457cd5dfdf88035c61ee19` in hxcpp,
that's the last commit that work for me.

**Comment by sumo961** [comment]
Thanks, Valentin. With 024c25f I was able to compile on a Mac.

Observations:

- the app is not recognized as such initially, I had to add ".app"

- However, the ".app" then appears in the app's name in the menu bar

- The app shows "Unknown image data format." when opened

- GL window is empty

- At the end of the compile process, the are many warnings about "ld:
warning: ld: warning: object file (obj/darwin64/3b09425d_TextCtrlStyle.o)
was built for newer OSX version (10.7) than being linked (10.5)object file
(obj/darwin64/3b09425d_WindowStyle.o) was built for newer OSX version
(10.7) than being linked (10.5)"
etc.. Is there anything I can do to fix this?
